uat/custom-components/src/main/java/com/aws/greengrass/artifacts/logGenerator.java

```diff
@@ -5,16 +5,200 @@
+        appender.start();
+
+        // set logger
+        logger.addAppender(appender);
+        logger.setLevel(Level.INFO);
+
+        return logger;
+    }
+
+    private static void performLogging(Logger logger, int totalLogNumbers, long logWriteFreqSeconds) {
+        for (int i=0; i < totalLogNumbers; i++) {
+            random = Math.random();
+            StringBuilder sb = generateLogMessage(i, logger.getName());
+            logger.info(sb.toString());
+            try {
+                TimeUnit.SECONDS.sleep(logWriteFreqSeconds);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    private static StringBuilder generateLogMessage(int seq, String loggerName) {
+        String[] namesArray = {"Jimmy", "Bob", "Amy", "I", "He", "She", "They"};
+        String[] verbsArray = {"play", "study", "develop", "know", "like"};
+        String[] nounsArray = {"code", "math", "football", "computer", "chemistry"};
+
+        int namesIndex = (int)(random * namesArray.length);
+        int verbsIndex = (int)(random * verbsArray.length);
+        int nounsIndex = (int)(random * nounsArray.length);
+        StringBuilder sb = new StringBuilder().
+                append(seq).
+                append("-" + loggerName).
+                append(" " + namesArray[namesIndex]).
+                append(" " + verbsArray[verbsIndex]).
+                append(" " + nounsArray[nounsIndex]);
+
+        return sb;
+    }
+
+    /*
+    Intentionally provide number of log files created in Device during the logging. This is only used for developers to
+    manually
+    check results in the local machine
+    */
+    private static void verifyLoggingSucceed(String currentPath, String fileName, String extension) {
+        File directory = new File(currentPath);
+        listMyFiles(directory);
+
+        File activeLogFile = new File(currentPath + "/" + fileName + "." + extension);
+        try {
+            if (activeLogFile.createNewFile()) {
+                System.out.println("new file created at "+ activeLogFile.getAbsolutePath());
+            } else {
+                System.out.println("file existed");
+            }
+        } catch (IOException e) {
+            System.exit(0);
+        }
+
+        // just try to verify if log messages are correctly written locally, instead of SSH to DUT devices. This
+        // should not be used as verification in UAT.
+        verifyFileIsWritten(activeLogFile);
+    }
+
+    private static void listMyFiles(File file) {
+        String[] pathnames = file.list();
+        System.out.println("listing files: " + pathnames.length);
+        for (String pathname : pathnames) {
+            System.out.println(pathname);
+        }
+    }
+
+    /*
+    This is only used to provide a way to read logs in file from local results, instead of SSH to the device.
+     */
+    private static void verifyFileIsWritten(File file) {
```

Comment:
This would run on your local machine or the machine you launched the tests from. Why is this needed?